libraries/chain/apply_context.cpp

```diff
@@ -237,6 +243,12 @@ void apply_context::require_recipient( account_name recipient ) {
          recipient,
          schedule_action( action_ordinal, recipient, false )
       );
+
+      if (eosio::chain::chain_config::deep_mind_enabled) {
```

Comment:
I notice that each time `dmlog` is called it is prefaced with this check.  Wouldn't it be clearer to move this check into `dmlog` itself?